### PR TITLE
feat: add jack analyzer

### PR DESCRIPTION
**Pull request by henok-tesfaye**
Make Progress on #32 

**What kind of change does this PR introduce?**
It's a feature PR which adds analyzer module for compiler construction.

**Does this PR introduce a breaking change?**
No

**What needs to be documented once your changes are merged?**
Nothing

### CODE REVIEW

**Comment by mezzzi** [comment]
@henok-tesfaye are you seeing this from your end?
Screenshot from 2020-11-06 11-54-55

**Comment by henok-outlier** [comment]
Nope @mezzzi .

Screen Shot 2020-11-06 at 11 58 07 AM

File: src/abstractions/software/compiler/mockData/index.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import SyntaxAnalayzerMockData from './syntaxAnalayzer'
```
Comment (mezzzi) [inline src/abstractions/software/compiler/mockData/index.js:1]:
```suggestion
import SyntaxAnalayzerMockData from './SyntaxAnalayzer'
```

**Review by mezzzi — CHANGES_REQUESTED** [review]
There is a minor typo in one of the imports

File: src/abstractions/software/compiler/mockData/index.js (hunk 1 above)
Comment (henok-outlier) [reply src/abstractions/software/compiler/mockData/index.js:1]: Is this issue causing the test to fail from your end?

Comment (henok-outlier) [reply src/abstractions/software/compiler/mockData/index.js:1]: And I'm not sure why it's working on my end.

Comment (mezzzi) [reply src/abstractions/software/compiler/mockData/index.js:1]: Yeah, when I fixed that typo the test run fine on my side

Comment (mezzzi) [reply src/abstractions/software/compiler/mockData/index.js:1]: Not sure why it didn't fail for you

Comment (henok-tesfaye) [reply src/abstractions/software/compiler/mockData/index.js:1]: Great, I'll fix it then. But curious to know why it's working with the current way.

**Review by mezzzi — APPROVED** [review]
Looks Great

**Pull request merged**